FizzBazz課題です。

yusukedayo opened the pull request:
確認よろしくお願いします。

yusukedayo committed:
FizzBazz課題です。

確認よろしくお願いします。

miketa-webprgr commented on practice.fizzbuzz.rb:
```diff
@@ -0,0 +1,14 @@
+number = ARGV[0].to_i
+p number
+    number.times do |x|
+        number += 1
+        if x % 15 == 0
+            puts "FizzBuzz"
+        elsif x % 5 == 0
+            puts "Buzz"
+        elsif x % 3 == 0
+            puts "Fizz"
+        else
+            puts x
+        end
+    end
```
ここであえてインデントするのは不要なので、`p number` と揃えておいてください。
また、流派があるので、押しつけはしないですが、rubyは基本的にはインデントを2 spacesとする人が多いです。

yusukedayo replied:
コメントありがとうございます！
インデントについてちゃんと考えていなかったので今後は気を付けます！

miketa-webprgr commented:
実行結果貼っておきます。
これは、要件とは異なりますよね？　

きちんと作ったプログラムは実行して、挙動が正しいか確認しましょう！
そして、違うということはどこかがおかしいので、修正を加えてみてください！

```
30
FizzBuzz
1
2
Fizz
4
Buzz
Fizz
7
8
Fizz
Buzz
11
Fizz
13
14
FizzBuzz
16
17
Fizz
19
Buzz
Fizz
22
23
Fizz
Buzz
26
Fizz
28
29
```

yusukedayo committed:
修正版です。

・インデントの変更
・任意の数よりも１少ない数までしか表示されなかった（30を入力したら29までの結果しか表示されない）ので修正しました。

yusukedayo commented:
修正版の方提出させてもらったので確認お願いします。
ご指摘頂いたように要件を満たしているかのチェックが甘かったので今後は気を付けます。

miketa-webprgr commented:
> 任意の数よりも１少ない数までしか表示されなかった（30を入力したら29までの結果しか表示されない）ので修正しました。

なるほど、そういうことですか。

```
30
FizzBuzz
```

で始まるので、すごく違和感がありましたが、ここはあくまで確認用？か何かだったんですかね。

こんなコードの方が結果に沿っているかと思いますが、とりあえずOKです！

```rb
number = ARGV[0].to_i
number.times do |x|
  x += 1
  number += 1
  if x % 15 == 0
    puts "FizzBuzz!"
  elsif x % 5 == 0
    puts "Buzz!"
  elsif x % 3 == 0
    puts "Fizz!"
  else
    puts x
  end
end
```

miketa-webprgr commented:
ちなみに、yusukeさんの実行結果も併せて貼っておきます。

```
31 　　　　　←　このあたりが少し違和感がありましたが、とりあえずOKです！
FizzBuzz!　　←　このあたりが少し違和感がありましたが、とりあえずOKです！
1
2
Fizz!
4
Buzz!
Fizz!
7
8
Fizz!
Buzz!
11
Fizz!
13
14
FizzBuzz!
16
17
Fizz!
19
Buzz!
Fizz!
22
23
Fizz!
Buzz!
26
Fizz!
28
29
FizzBuzz!
```

yusukedayo committed:
修正版#2です。

ご指摘頂いた部分を修正しました。
p numberに関しては作成途中で確認していたものがそのままになってしまっていました。
指摘してもらう前に気づけるように努力します！
